keps/sig-architecture/2527-clarify-status-observations-vs-rbac/README.md

```diff
@@ -0,0 +1,697 @@
+point in the project, I expect that wiping status out like this would end in
+fireworks, and not the fun kind.  Status is, effectively, as durable as spec.
+
+Many of our APIs pass this test (sometimes we fudge it and say yes "in
+theory"), but not all of them do.  This KEP proposes to clarify or remove this
+guidance, especially as it pertains to state that is not derived from
+observation.
+
+One of the emergent uses of the spec/status split is access control.  It is
+assumed that, for most resources, users own (can write to) all of spec and
+controllers own all of status, and not the reverse.  This allows patterns like
+Services which set `spec.type: LoadBalancer`, where the controller writes the
+LB's IP address to status, and kube-proxy can trust that IP address (because it
+came from a controller, not a user).  Compare that with Services which use
+`spec.externalIPs`.  The behavior in kube-proxy is roughly the same, but
+because non-trusted users can write to `spec.externalIPs` and that does not
+require a trusted controller to ACK, that behavior was declared a CVE.
+
+This KEP further proposes to add guidance for APIs that want to implement an
+"allocation" or "binding" pattern which requires trusted ACK.
+
+## Motivation
+
+As an API reviewer, I have seen many different patterns in use.  I have shot
+down APIs that run afoul of the rebuild-from-observation test, and forced the
+implementations to be more complicated.  I no longer find this to be useful to
+our APIs, and in fact I find it to be a detriment.
+
+I suspect that several APIs would have come out differently if not for this.
+
+### Goals
+
+* Clarify or remove the from-observation rule to better match reality.
+* Provide guidance for APIs on how to save controller results.
+
+### Non-Goals
+
+* Retroactively apply this pattern to change existing GA APIs.
+* Necessarily apply this pattern to change pre-GA APIs (though they may choose
+  to follow it).
+* Provide arbitrary storage for controller state.
+* Provide distinct access control for subsets of spec or status.
+
+## Proposal
+
+This KEP proposes to:
+
+1) Get rid of the [if it got lost](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status)
+litmus test for status fields
+2) Acknowledge that spec/status are a useful and meaningful split for access control
+3) Document one or more patterns for APIs that need a trusted acknowledgement
+   of the spec
+
+Depending on feedback, there may be other approaches to solving the same
+problems.
+
+### Risks and Mitigations
+
+The current guidance exists for a reason.  It does encourage a sort of
+idealogical cleanliness.  However, it is not universally adhered to and it
+neglects the reality of these request/acknowledge APIs.  On the whole, this KEP
+posits that the current guidance is a net negative.
+
+## Design Details
+
+<<[UNRESOLVED ideas to debate]>>
+
+### Option 1: Loosen status
```

Comment:
Please change this to just be the design, and options 2 and 3 to be "rejected alternatives" :)

Reply:
+1 the cat already left the barn, cows out of the bag etc; we may be able to hold some sort of line for the core, but wider out in the ecosystem, it's diminishingly unlikely resources can keep status pure.